### PR TITLE
KOGITO-8371: Update Cypress version to 12.2.0

### PR DESCRIPTION
**Pull request by tomasdavidorg**
Signed-off-by: Tomas David <tdavid@redhat.com>

https://issues.redhat.com/browse/KOGITO-8371

* update cypress to version 12.2.0 and cypress-real-events to 1.7.4
* update and merge kie-editors-standalone tests to be isolated from each other
* fix boxed-expression-component tsconfig.json to correctly include cypress namespace in vscode
* boxed-expression-component it-tests were changed to be isolated from each other
* fixed boxed-expression-component tsconfig.json to correctly include cypress n
* remove cypress commands and dependencies from packages which don't have cypress tests:
  * dashbuilder-editor
  * dashbuilder-viewer
  * import-java-classes-component
  * serverless-workflow-combined-editor
  * serverless-workflow-mermaid-viewer
  * serverless-workflow-standalone-editor
  * serverless-workflow-text-editor
  * text-editor
  * yard-editor

### CODE REVIEW

**Review by paulovmr — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts [hunk 1]
```diff
@@ -15,276 +15,226 @@
...
-      });
 
-      it("Process documentation property is read only", () => {
-        cy.wrap(propertyItems)
-          .find("textarea[name*='diagramSet.documentation']")
-          .should("have.value", "Documentation")
-          .should("not.have.class", "editable");
-      });
+        // Process documentation property is read only
+        cy.wrap($properties).find("textarea[name*='diagramSet.documentation']").should("have.value", "Documentation");
```
Comment (barboras7) [inline packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]: Is `.should("not.have.class", "editable")` skipped on purpose (compared to original code)?

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts [hunk 2]
```diff
@@ -15,276 +15,226 @@
...
           .find("input[id='importsTextBox']")
           .should("have.value", "No imports")
           .should("not.have.class", "editable");
-      });
 
-      it.skip("Process imports buttons is read only", () => {
-        cy.wrap(propertyItems).find("button[id='importsButton']").should("be.disabled");
-      });
+        // Process imports buttons is read only - is currently enabled
+        // cy.wrap($properties).find("button[id='importsButton']").should("be.disabled");
```
Comment (barboras7) [inline packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]: IMHO, the tests that were skipped in the original code anyway could be just removed. I think that no one will ever go searching for skipped tests and enable them later, why should we keep such code in the code base then. WDYT?

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts [hunk 3]
```diff
@@ -15,276 +15,226 @@
...
+          .then(($globalVariablesContainer) => {
+            cy.wrap($globalVariablesContainer).find("#variableRow").siblings().should("have.length", "0");
+
+            // Process global variable name is read only
+            cy.wrap($globalVariablesContainer)
+              .find("input[data-field='name']")
+              .should("not.have.class", "editable")
+              .should("have.value", "is_processing");
+
+            // Process global variable typeis read only"
```
Comment (barboras7) [inline packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]:
```suggestion
            // Process global variable type is read only"
```

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-workitem.cy.ts [hunk 4]
```diff
@@ -84,173 +84,141 @@ describe("Bpmn Workitem E2E Test.", () => {
...
+        // Data assignments of custom work item tasks are not disabled", () => {
+        cy.wrap($properties)
+          .find("a")
+          .contains("Data Assignments")
+          .should("not.be.disabled")
+          .click()
+          .wait(1000)
+          .scrollIntoView();
+
+        // Custom workitem task has expected data inputs and data output assignments
```
Comment (barboras7) [inline packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-workitem.cy.ts]: There is no relevant testing method below this description

Comment (tomasdavidorg) [reply packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-workitem.cy.ts]: Moved to correct position.

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts (hunk 3 above)
Comment (tomasdavidorg) [reply packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]: Fixed.

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts (hunk 2 above)
Comment (tomasdavidorg) [reply packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]: Removed.

File: packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts (hunk 1 above)
Comment (tomasdavidorg) [reply packages/kie-editors-standalone/it-tests/cypress/e2e/bpmn-read-only.cy.ts]: Thanks. I missed this. It is added back.

**Review by barboras7 — APPROVED** [review]
LGTM, thanks @tomasdavidorg!

**Pull request merged**